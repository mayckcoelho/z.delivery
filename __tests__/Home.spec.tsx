import 'jsdom-global/register'
import React from 'react'
import '../config/setup'
import Home from '../src/containers/home'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { unmountComponentAtNode } from 'react-dom'

let container: HTMLDivElement | null = document.createElement('div')

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/',
    search: '',
    hash: '',
    state: null,
  }),
  useHistory: () => ({
    push: jest.fn(),
  }),
}))

beforeEach(() => {
  container && document.body.appendChild(container)
})

afterEach(() => {
  container && unmountComponentAtNode(container)
  container?.remove()
  container = null
})

describe('Home should render correctly', () => {
  it('snapshot match', () => {
    const home = shallow(<Home />)

    expect(toJson(home)).toMatchSnapshot()
  })

  it('should mount component', () => {
    mount(<Home />)
  })
})
