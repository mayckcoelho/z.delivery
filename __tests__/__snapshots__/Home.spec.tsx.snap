// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly snapshot match 1`] = `
<main>
  <section
    className="background"
  >
    <div
      className="background__wrapper"
    >
      <h1
        className="background__wrapper__title"
      >
        Informe seu endereço e nós vamos encontrar as
         
        <strong>
          melhores ofertas
        </strong>
      </h1>
      <Home
        handleAction={[Function]}
      />
    </div>
  </section>
  <Resume />
  <Hero />
</main>
`;
